src/main/java/com/example/demo/model/ShoppingCart.java

```diff
@@ -0,0 +1,41 @@
+package com.example.demo.model;
+
+import jakarta.persistence.CascadeType;
+import jakarta.persistence.Column;
+import jakarta.persistence.Entity;
+import jakarta.persistence.FetchType;
+import jakarta.persistence.GeneratedValue;
+import jakarta.persistence.GenerationType;
+import jakarta.persistence.Id;
+import jakarta.persistence.OneToMany;
+import jakarta.persistence.OneToOne;
+import jakarta.persistence.PrimaryKeyJoinColumn;
+import jakarta.persistence.Table;
+import java.util.Set;
+import lombok.EqualsAndHashCode;
+import lombok.Getter;
+import lombok.Setter;
+import lombok.ToString;
+import org.hibernate.annotations.SQLDelete;
+import org.hibernate.annotations.SQLRestriction;
+
+@Entity
+@Getter
+@Setter
+@SQLDelete(sql = "UPDATE shopping_cart SET is_deleted = true WHERE id = ?")
+@SQLRestriction(value = "is_deleted = false")
+@Table(name = "shopping_cart")
```

Comment:
```suggestion
@Table(name = "shopping_carts")
```
According to the naming convention, table name should be plural